x-pack/plugins/search_playground/public/hooks/use_indices_validation.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useState } from 'react';
+import { useQueryIndices } from './use_query_indices';
+
+export const useIndicesValidation = (unvalidatedIndices: string[]) => {
+  const [isValidated, setIsValidated] = useState<boolean>(false);
+  const [validIndices, setValidIndices] = useState<string[]>([]);
+  const { indices, isFetched: isIndicesLoaded } = useQueryIndices();
+
+  useEffect(() => {
+    if (isIndicesLoaded) {
+      setValidIndices(indices.filter((index) => unvalidatedIndices.includes(index)));
```

Comment:
useQueryIndices gets the top 10 indices, what happens if its not in this list but does exist?

I would do an explicit call for all the unvalidatedIndices to see if they exist.

Reply:
https://www.elastic.co/guide/en/elasticsearch/reference/current/cat-indices.html supports a target.